test/jdk/javax/swing/DataTransfer/ListDragOverFeedbackTest.java

```diff
@@ -0,0 +1,119 @@
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual ListDragOverFeedbackTest
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.GridLayout;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import javax.swing.BorderFactory;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JPanel;
+import javax.swing.TransferHandler;
+
+public class ListDragOverFeedbackTest {
+    private static final String INSTRUCTIONS = """
+        JList should provide visual feedback when a DnD drag operation is
+        occurring over it. This test is to check that it provides the
+        feedback about the drop location correctly.
+
+        Click on the label where it says "DRAG FROM HERE" and begin dragging.
+        Drag over each column in each of the three JLists and make sure that
+        the drop location indicated is appropriate for the mouse location. For
+        instance, if the mouse is in the first column, the drop location should
+        also be indicated in that first column.
+
+        If above is true press PASS else press FAIL.
+        """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .columns(50)
+                .testUI(ListDragOverFeedbackTest::createTestUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static final TransferHandler handler = new TransferHandler() {
+        public boolean canImport(JComponent comp, DataFlavor[] flavors) {
+            return true;
+        }
+    };
+
+    private static JFrame createTestUI() {
+        String[] vals = new String[] {
+                "one", "two", "three", "four", "five", "six", "seven", "eight",
+                "nine", "ten", "eleven", "twelve", "thirteen", "fourteen"};
+
+        JFrame frame = new JFrame("ListDragOverFeedbackTest");
+        final JLabel label = new JLabel("DRAG FROM HERE");
+        label.setPreferredSize(new Dimension(400, 25));
+        label.setTransferHandler(new TransferHandler("text"));
+        label.addMouseListener(new MouseAdapter() {
+            public void mousePressed(MouseEvent me) {
+                label.getTransferHandler().exportAsDrag(label, me,
+                                              TransferHandler.COPY);
+            }
+        });
+
+        JList list1 = new JList(vals);
+        list1.setTransferHandler(handler);
+        list1.setBorder(BorderFactory.createLineBorder(Color.BLACK));
+
+        JList list2 = new JList(vals);
+        list2.setLayoutOrientation(JList.VERTICAL_WRAP);
+        list2.setTransferHandler(handler);
+        list2.setBorder(BorderFactory.createLineBorder(Color.BLACK));
```

Comment:
The first two lists seem to be cut off on MacOS at least. The test itself works and passes.

<img width="392" alt="Screenshot 2025-04-16 at 1 46 16 PM" src="https://github.com/user-attachments/assets/b452ed69-cc4a-4bb6-9cd1-c5ecce6bcec6" />

Reply:
Still slightly cut off for the first list. But I guess that one doesn't matter as much since 10, 11, 12, 13, and 14 aren't even visible. That one is supposed to have no wrap too. LGTM now

<img width="392" alt="Screenshot 2025-04-16 at 2 21 03 PM" src="https://github.com/user-attachments/assets/82af89b0-8034-4907-8555-d26f273a6300" />